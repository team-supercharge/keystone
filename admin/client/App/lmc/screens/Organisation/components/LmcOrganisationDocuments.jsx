import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ActionCreators } from '../../../actions/actions'
import { connect } from 'react-redux'
import { BlankState } from '../../../../elemental'
import LmcCreateButton from '../../../components/LmcCreateButton.jsx'

export class LmcOrganisationDocuments extends Component {
    renderDocuments = () => {
        return (
            <div>
                { JSON.stringify(this.props.documents) }
            </div>
        )
    }

    render() {
        const { documents, fetchDocuments } = this.props
        const hasDocuments = documents && documents.length

        return (
            <div style={styles.mainContainer}>
                <LmcCreateButton
                    buttonText='Document'
                    title='Add a Document'
                    listId='HomeDocument'
                    onCreate={fetchDocuments}
                    style={styles.createButton}
                />
                { hasDocuments ? (
                    <div>
                        { this.renderDocuments() }
                    </div>
                ) : (
                    <BlankState
                        heading={NO_DOCUMENTS_MESSAGE}
                        style={styles.noDocumentsMessage}
                    />
                )}
            </div>
        )
    }
}

const NO_DOCUMENTS_MESSAGE = "You haven't added any documents for your home yet"

const styles = {
    createButton: {
        float: 'right',
        width: 200,
    },
    mainContainer: {
        
    },
    noDocumentsMessage: {

    }
}

LmcOrganisationDocuments.propTypes = {
    documents: PropTypes.array,
    fetchDocuments: PropTypes.func.isRequired
}

const mapStateToProps = state => {
    return {
        documents: state.data['home-documents']
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchDocuments: () => dispatch(ActionCreators.loadList('home-documents'))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LmcOrganisationDocuments)