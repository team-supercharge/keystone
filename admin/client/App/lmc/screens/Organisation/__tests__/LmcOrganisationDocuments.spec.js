import React from 'react'
import { shallow } from 'enzyme'
import { LmcOrganisationDocuments } from '../components/LmcOrganisationDocuments.jsx'

describe('LmcOrganisationDocuments', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(
            <LmcOrganisationDocuments />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})