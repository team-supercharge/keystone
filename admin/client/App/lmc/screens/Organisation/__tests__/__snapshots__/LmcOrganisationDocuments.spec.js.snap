// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcOrganisationDocuments renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "height": "85vh",
        "overflow": "scroll",
        "padding": "50px 20px 0px 20px",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 1000,
        }
      }
    >
      <LmcCreateButton
        buttonText="Document"
        listId="HomeDocument"
        onCreate={[MockFunction]}
        style={
          Object {
            "float": "right",
            "width": "15vw",
          }
        }
        title="Add a Document"
      />
      <LmcDocumentList
        documents={
          Object {
            "Policies": Array [
              Object {
                "name": "TestDocument",
              },
            ],
            "Procedures": Array [
              Object {
                "name": "TestDocument2",
              },
            ],
          }
        }
        listId="home-documents"
        onDelete={[MockFunction]}
      />
    </div>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "height": "85vh",
          "overflow": "scroll",
          "padding": "50px 20px 0px 20px",
        }
      }
    >
      <div
        style={
          Object {
            "margin": "0 auto",
            "maxWidth": 1000,
          }
        }
      >
        <LmcCreateButton
          buttonText="Document"
          listId="HomeDocument"
          onCreate={[MockFunction]}
          style={
            Object {
              "float": "right",
              "width": "15vw",
            }
          }
          title="Add a Document"
        />
        <LmcDocumentList
          documents={
            Object {
              "Policies": Array [
                Object {
                  "name": "TestDocument",
                },
              ],
              "Procedures": Array [
                Object {
                  "name": "TestDocument2",
                },
              ],
            }
          }
          listId="home-documents"
          onDelete={[MockFunction]}
        />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcOrganisationDocuments
        deleteDocument={[MockFunction]}
        documents={
          Object {
            "Policies": Array [
              Object {
                "name": "TestDocument",
              },
            ],
            "Procedures": Array [
              Object {
                "name": "TestDocument2",
              },
            ],
          }
        }
        fetchDocuments={[MockFunction]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcOrganisationDocuments {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "deleteDocument": [MockFunction],
          "documents": Object {
            "Policies": Array [
              Object {
                "name": "TestDocument",
              },
            ],
            "Procedures": Array [
              Object {
                "name": "TestDocument2",
              },
            ],
          },
          "fetchDocuments": [MockFunction],
        },
        "refs": Object {},
        "state": Object {
          "documentsFetchInterval": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "height": "85vh",
              "overflow": "scroll",
              "padding": "50px 20px 0px 20px",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 1000,
              }
            }
          >
            <LmcCreateButton
              buttonText="Document"
              listId="HomeDocument"
              onCreate={[MockFunction]}
              style={
                Object {
                  "float": "right",
                  "width": "15vw",
                }
              }
              title="Add a Document"
            />
            <LmcDocumentList
              documents={
                Object {
                  "Policies": Array [
                    Object {
                      "name": "TestDocument",
                    },
                  ],
                  "Procedures": Array [
                    Object {
                      "name": "TestDocument2",
                    },
                  ],
                }
              }
              listId="home-documents"
              onDelete={[MockFunction]}
            />
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "height": "85vh",
              "overflow": "scroll",
              "padding": "50px 20px 0px 20px",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 1000,
              }
            }
          >
            <LmcCreateButton
              buttonText="Document"
              listId="HomeDocument"
              onCreate={[MockFunction]}
              style={
                Object {
                  "float": "right",
                  "width": "15vw",
                }
              }
              title="Add a Document"
            />
            <LmcDocumentList
              documents={
                Object {
                  "Policies": Array [
                    Object {
                      "name": "TestDocument",
                    },
                  ],
                  "Procedures": Array [
                    Object {
                      "name": "TestDocument2",
                    },
                  ],
                }
              }
              listId="home-documents"
              onDelete={[MockFunction]}
            />
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcOrganisationDocuments
    deleteDocument={[MockFunction]}
    documents={
      Object {
        "Policies": Array [
          Object {
            "name": "TestDocument",
          },
        ],
        "Procedures": Array [
          Object {
            "name": "TestDocument2",
          },
        ],
      }
    }
    fetchDocuments={[MockFunction]}
  />,
}
`;
